src/libraries/Common/src/System/Reflection/TypeNameParser.Helpers.cs

```diff
@@ -0,0 +1,157 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+
+#nullable enable
+
+namespace System.Reflection
+{
+    internal partial struct TypeNameParser
+    {
+#if NETCOREAPP
+        private static ReadOnlySpan<char> CharsToEscape => "\\[]+*&,";
```

Comment:
I think we could use .NET 8's `SearchValues<char>` here?  e.g.
```suggestion
#if NET8_0_OR_GREATER
        private static ReadOnlySpan<char> CharsToEscape => "\\[]+*&,";
#else
        private static readonly SearchValues<char> CharsToEscape = SearchValues.Create("\\[]+*&,");
#endif
```